scripts/monitoring/auto_manage_ec2_workers.py

```diff
@@ -0,0 +1,172 @@
+    "AWS_ACCESS_KEY_ID": os.environ.get("AWS_ACCESS_KEY_ID", "x"),
+    "AWS_SECRET_ACCESS_KEY": os.environ.get("AWS_SECRET_ACCESS_KEY", "x"),
+    "AWS_REGION": os.environ.get("AWS_DEFAULT_REGION", "us-east-1"),
+    "AWS_STORAGE_BUCKET_NAME": os.environ.get(
+        "AWS_STORAGE_BUCKET_NAME", "evalai-s3-bucket"
+    ),
+}
+
+
+ENV = os.environ.get("ENV", "dev")
+
+evalai_endpoint = os.environ.get("API_HOST_URL", "http://localhost:8000")
+auth_token = os.environ.get(
+    "AUTH_TOKEN",
+)
+
+
+def get_boto3_client(resource, aws_keys):
+    client = boto3.client(
+        resource,
+        region_name=aws_keys["AWS_REGION"],
+        aws_access_key_id=aws_keys["AWS_ACCESS_KEY_ID"],
+        aws_secret_access_key=aws_keys["AWS_SECRET_ACCESS_KEY"],
+    )
+    return client
+
+
+def get_pending_submission_count(challenge_metrics):
+    pending_submissions = 0
+    for status in ["running", "submitted", "queued", "resuming"]:
+        pending_submissions += challenge_metrics.get(status, 0)
+    return pending_submissions
+
+
+def stop_workers(challenge):
+    target_instance_name = "Worker-Instance-{}-{}".format(
+        ENV, challenge["id"]
+    )
+
+    ec2 = get_boto3_client("ec2", aws_keys)
+    response = ec2.describe_instances(
+        Filters=[
+            {"Name": "tag:Name", "Values": [target_instance_name]},
+            {"Name": "instance-state-name", "Values": ["running"]},
+        ]
+    )
+
+    instances = [
+        instance
+        for reservation in response["Reservations"]
+        for instance in reservation["Instances"]
+    ]
+
+    if instances:
+        for instance in instances:
+            instance_id = instance["InstanceId"]
+            response = ec2.stop_instances(InstanceIds=[instance_id])
+            print("AWS API Response: {}".format(response))
+            print(
+                "Stopped worker for Challenge ID: {}, Title: {}".format(
+                    challenge["id"], challenge["title"]
+                )
+            )
+    else:
+        print(
+            "No running instances with the specified name found for Challenge ID: {}, Title: {}. Skipping.".format(
+                challenge["id"], challenge["title"]
+            )
+        )
+
+
+def start_workers(challenge):
+    target_instance_name = "Worker-Instance-{}-{}".format(
+        ENV, challenge["id"]
+    )
+
+    ec2 = get_boto3_client("ec2", aws_keys)
+    response = ec2.describe_instances(
+        Filters=[
+            {"Name": "tag:Name", "Values": [target_instance_name]},
+            {"Name": "instance-state-name", "Values": ["stopped"]},
+        ]
+    )
+
+    instances = [
+        instance
+        for reservation in response["Reservations"]
+        for instance in reservation["Instances"]
+    ]
+
+    if instances:
+        for instance in instances:
+            instance_id = instance["InstanceId"]
+            response = ec2.start_instances(InstanceIds=[instance_id])
```

Comment:
Same thing here. Also, if multiple instances are created with same name due to approving and unapproving several times, this will start all the instances which is wrong.